Documentation/installation-with-spack.rst

```diff
@@ -0,0 +1,391 @@
+present in your system, you can install it with Spack. For example, let's 
+install *gcc 8.3.0*:
+
+.. code-block:: bash
+
+  spack install gcc@8.3.0
+
+
+Don't forget to add it to Spack once it is installed:
+
+.. code-block:: bash
+
+  spack compiler find $(spack location -i gcc@8.3.0)
+
+
+Type the following to see all compilers avaliable for Spack
+
+.. code-block:: bash
+
+  spack compiler list
+
+
+Environment Modules
+-------------------
+
+You can install environment modules to be able to *load* and *unload*
+packges, libraries and compilers installed with Spack. 
+
+.. code-block:: bash
+
+  spack bootstrap
+
+
+After that you can work with the installed software as following:
+
+.. code-block:: bash
+
+  module avail
+  module load <package name>
+  module list
+  module unload <package>
+  module purge
+
+You can also look at the list of installed software as following:
+
+.. code-block:: bash
+
+  # the most concise list
+  spack find
+
+  # a list of packages with options requested during their instalation
+  spack find -v
+
+  # the most detailed list (including install-options of all packages and their deps.)
+  spack find -v -d
+
+
+SeisSol-Env
+-----------
+
+The purpose of the script is to install essential packages and libraries for 
+SeisSol as well as to install some extra, optional packages that might 
+be useful. Here is a list of the essentials:
+
+- hdf5, version=1.8.21
+- netcdf-c, version=4.4.0
+- libxsmm, version=latest
+- pspamm
+- memkind, version=latest
+
+
+*NOTE*: **python3**, **numpy** and **scipy** also belong to the essential 
+set and must be on your system to be able to compile SeisSol. However, they 
+do not affect run-time performance of SeisSol and most the UNIX-based systems 
+have these packages pre-installed. Therefore, installation of these packages 
+are optional to save the set-up time. You can trigger an installation of 
+them if your system comes without python3 (version=3.5.2), numpy and 
+scipy (see, examples). We use the same strategy and reasoning for **cmake** and 
+**scons**.
+
+Additionally, a user can customize each individual dependency using 
+Spack 
+`recursive syntax <https://spack.readthedocs.io/en/latest/basic_usage.html#specs-dependencies>`_. 
+
+
+Options
+~~~~~~~
+
+- *asagi* [**default=on**, off] - installs asagi 
+- *building_tools* [**default=on**, off] - installs scons and cmake
+- *extra_blas* [**default=none**, mkl, openblas, blis] - installs extra blas implementations
+- *mpi* [**default=on**, off] - installs an MPI implementation
+- *python* [on, **default=off**] - installs python, numpy, scipy and pip
+
+*NOTE*: mpi is a virtual package, a user must specify a concrete implementation
+of the standard
+
+Examples
+~~~~~~~~
+
+.. code-block:: bash
+
+  # 1. with intel compiler suite
+  spack install seissol-env +mpi +asagi ^intel-mpi@2018.2.199 %intel@17.0.2
+
+  # 2. with gcc compiler suite
+  spack install seissol-env +mpi +asagi ^openmpi@3.1.5 %gcc@8.3.0
```

Comment:
When I execute this line (tried on LRZ workstation and personal notebook) I get the following error:
seissol-env requires cmake version 3.12.0:3.16.2, but spec asked for 3.16.5
I have installed cmake 3.15.2 and 3.16.4 respectively.